Doppler.Sap.Test/Utils/HttpHelperExtension.cs

```diff
@@ -0,0 +1,33 @@
+using Moq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq.Protected;
+
+namespace Doppler.Sap.Test.Utils
+{
+    public static class HttpHelperExtension
+    {
+        public static IHttpClientFactory GetHttpClientMock(string content, HttpStatusCode statusCode, Mock<HttpMessageHandler> httpHandlerMock = null)
+        {
+            var httpClientFactoryMock = new Mock<IHttpClientFactory>();
+            var httpMessageHandlerMock = httpHandlerMock ?? new Mock<HttpMessageHandler>();
```

Comment:
If I understand it well, you are accepting a `Mock<HttpMessageHandler>` or creating a new one.

But, in both cases, you are manipulating it setting up `SendAsync` behavior. In my opinion, it is broken the [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

Do you really need both options?

Reply:
You are right, I don't need both options, only I need that the `Mock<HttpMessageHandler>` should be created in the test for `assert ` purposes, I'll change it thanks